tests/test_integration.py

```diff
@@ -89,10 +91,48 @@ def test_good_result_manifests():
     """
     result = run_query("test_all_result_groups", Namespace())
 
-    for group, count in result["data"]:
+    for group, _ in result["data"]:
         if group is None:
             continue
 
         assert (
             not task.is_bad_group("x", group) and "\\" not in group
         ), f"{group} group is bad!"
+
+
+def test_caching_of_tasks(tmp_path):
+    config_file = tmp_path / "config.toml"
+    text = (
+        """
+[adr]
+verbose = true
+[adr.cache]
+default = "file"
+retention = 1000
+[adr.cache.stores]
+file = { driver = "file", path = "%s" }
+"""
+        % tmp_path
+    )
+    config_file.write_text(text)
+    config = Configuration(path=config_file)
+
+    # Once we reach Nov. 23rd, 2020 the test should
+    TASK_ID = "WGNh9Xd8RmSG_170-0-mkQ"
+    # Making sure there's nothing left in the cache
+    assert config.cache.get(TASK_ID) is None
+    # Push from Nov. 22nd, 2019
+    push = Push("6e87f52e6eebdf777f975baa407d5c22e9756e80", branch="mozilla-beta")
+    tasks = push.tasks()
+    for t in tasks:
+        if t.id == TASK_ID:
```

Comment:
I'd make sure we actually do find the ID in the tasks, otherwise if we introduce a regression somewhere the test might still pass

Reply:
Handled with `assert found_task`.